### PR TITLE
Various linting scripts

### PR DESCRIPTION
**Pull request by skjnldsv**
Everything is put under the same test suite name:
- `Lint / eslint`
- `Lint / stylelint`
- `Lint / php`
- `Lint / php-cs`

### CODE REVIEW

File: workflow-templates/lint-php.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set up php ${{ matrix.php-versions }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-versions }}
+          coverage: none
+
+      - name: Lint
+        run: composer run lint
```
Comment (skjnldsv) [inline workflow-templates/lint-php.yml:35]: Should we have the script hardcoded here or rely on the composer.json?

File: workflow-templates/lint-eslint.properties.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Linting javascript",
+    "description": "Javascript linting. Require proper script in package.json",
```
Comment (ChristophWurst) [inline workflow-templates/lint-eslint.properties.json:3]: what is proper?

Comment (skjnldsv) [reply workflow-templates/lint-eslint.properties.json:3]: That is the question!
Same for the composer entries.
Should we have the script hardcoded here or rely on the composer.json?

**Comment by ChristophWurst** [comment]
if we document the convention somewhere similar to https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/npm.html#npm-run-lint-optional then I'm fine with a hardcoded default

File: workflow-templates/lint-eslint.properties.json (hunk 2 above)
Comment (skjnldsv) [reply workflow-templates/lint-eslint.properties.json:3]: I guess we don't/can't have templates for what to have in the XXX.json configs, do you have any idea on how to approach this?

Comment (skjnldsv) [reply workflow-templates/lint-eslint.properties.json:3]: https://github.com/nextcloud/documentation/pull/6930

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**